complex/cargo/store/src/lib.rs

```diff
@@ -40,10 +40,14 @@ impl Drop for Store {
 }
 
 impl Store {
-    pub fn new(uri: String) -> Self {
+    pub fn new(uri: Option<String>) -> Self {
+        let c = match &uri {
+            &Some(ref u) => Connection::open(u.clone()).unwrap(),
```

Comment:
N.B., that `unwrap` here is non-descriptive and panics. Prefer explicit matching on the output, or if you're OK crashing, use `expect` with a message.